jni4net.n/src/jni/JNI.cs

```diff
@@ -68,146 +67,6 @@ private static void Init()
             }
         }
 
-        private static string FindJvmDir()

```

Comment:
I guess this will be necessary of windows.

Reply:
I thought that it's common practice to set JAVA_HOME on windows.

I will revert this change

Reply:
I started working with this when Java was between 1.5 and 1.8 in the wild. Server and client versions had different layouts on disk. Also there was 32bit version in different folder. Maybe this doesn't matter anymore ?

Also, when you deploy it to environments that you don't own, JAVA_HOME could be missing or wrong. At some point there was also code reading windows registry to figure it out.

When I'm remembering about deployments, I should also share that J2EE is horror story. Specifically it's class loaders are mess.

Anyway, it would be good create some document and list "goals and non-goals" with regards to supported combinations/versions.

Reply:
As far as I see, through the years it seems it gets better: JDK, OpenJDK, Microsoft compiled OpenJDK, Redhat provided OpenJDK, JRE etc 

There are 2 ways to see it:
- It's too complicated to catch all possible scenarios, so since we are a library and not an installer, let's keep it clean and let the caller set the JAVA_HOME or Bridge.JavaHome properly. It's not our problem.
- Or we replicate the search logic of some other library or eclipse (there are various related issues on github)

I don't have preference, I will have a look for some best effort algorithm and resolve this.

> Anyway, it would be good create some document and list "goals and non-goals" with regards to supported combinations/versions.

Do you  mean supported java versions? So far all test pass on Java 11, 17 and 21 on x64 windows and linux. I will check ARM at some point, but I don't think there would be any issue